dart/lib/src/http_client/sentry_http_client.dart

```diff
@@ -148,3 +148,12 @@ class SentryStatusCode {
     return '$_min..$_max';
   }
 }
+
+extension SentryStatusCodeListExtension on List<SentryStatusCode> {
+  bool containsStatusCode(int? statusCode) {
```

Comment:
Somewhat unrelated, but can we somehow use the `ValidateStatus` from the Dio options as default for `failedRequestStatusCodes`?

Reply:
I'm thinking of removing the `FailingRequestAdapter` all together because of this. If Dio throws on specific status codes anyway, there's no need for that functionality.

Reply:
https://github.com/getsentry/sentry-dart/pull/728